### PR TITLE
feat(sqlite)!: move sqlite to the `@sequelize/sqlite` package 

### PR DESCRIPTION
**Pull request by lohart13**
## Goal of the PR

Related issue: #13722
Related PR: #17190 

This PR moves the SQLite Dialect to the `@sequelize/sqlite` package.

## Relevant impacts

- This package pre-installs the necessary sqlite-specific dependency `sqlite`. Users won't need to install the additional packages anymore. This gives us control over which version we support.
- Unlike the postgres package, this package does not have unit tests yet. Once we add those, the script should be added to the package package.json and CI.

## Other changes

- Add ESM named exports tests for `@sequelize/mariadb` and `@sequelize/mysql`.

## Breaking changes

- Instead of installing `sqlite` users need to install `@sequelize/sqlite`.

### CODE REVIEW

File: packages/core/src/sequelize.js [hunk 1]
```diff
@@ -873,6 +872,8 @@ Use Sequelize#query if you wish to use replacements.`);
    */
   async _syncModelsWithCyclicReferences(options) {
     if (this.dialect.name === 'sqlite') {
+      // eslint-disable-next-line import/no-extraneous-dependencies -- only import when needed
+      const { withSqliteForeignKeysOff } = await import('@sequelize/sqlite');
```
Comment (lohart13) [inline packages/core/src/sequelize.js:876]: This is not ideal however if the `@sequelize/sqlite` package becomes a dependency, we end up with having cyclic dependencies.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -90,7 +90,7 @@ jobs:
         run: yarn lerna run test-unit-postgres --scope=@sequelize/core
       - name: Unit tests (postgres package)
         run: yarn lerna run test-unit --scope=@sequelize/postgres
-      - name: Unit tests (sqlite)
+      - name: Unit tests (ore - sqlite)
```
Comment (WikiRik) [inline .github/workflows/ci.yml:93]:
```suggestion
      - name: Unit tests (core - sqlite)
```

File: packages/core/src/sequelize.js (hunk 1 above)
Comment (WikiRik) [reply packages/core/src/sequelize.js:876]: We can move this function to the core utils folder for now and figure out a better way later

File: packages/sqlite/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  "type": "commonjs",
+  "version": "0.0.0-development",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@sequelize/core": "workspace:*",
+    "@sequelize/utils": "workspace:*",
+    "lodash": "4.17.21",
+    "sqlite3": "5.1.7"
```
Comment (WikiRik) [inline packages/sqlite/package.json:39]:
```suggestion
    "lodash": "^4.17.21",
    "sqlite3": "^5.1.7"
```

File: packages/core/src/sequelize.js (hunk 1 above)
Comment (lohart13) [reply packages/core/src/sequelize.js]: Updated now.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (lohart13) [reply .github/workflows/ci.yml]: Updated now

File: packages/sqlite/package.json (hunk 3 above)
Comment (lohart13) [reply packages/sqlite/package.json]: Updated now

**Review by ephys — APPROVED** [review]

**Pull request merged**